modules/createMapToProps.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    MapToProps,
+    MapStateToProps,
+    MapStateCreatorsToProps,
+    SetState
+} from './types'
+
+export default function bindMapStateToProps<P, S, ExtraP>(
+    initialProps: P,
+    mapStateToProps?:
+        | MapStateToProps<P, S, ExtraP>
+        | MapStateCreatorsToProps<S, ExtraP>
+): MapToProps<P, S, ExtraP> {
+    if (typeof mapStateToProps === 'function') {
+        return mapStateToProps(initialProps)
```

Comment:
This means the user has to explicitly spread `state` in `mapStateToProps`. Alternative:

```js
if (typeof mapStateToProps === 'function') {
    const map = mapStateToProps(initialProps) as Function

    const mapToProps = (state, setState) => ({
        ...state,
        ...map(state, setState)
    })

    return mapToProps as MapToProps<P, S, ExtraP>
}
```

Worth it? 🤔

Reply:
Maybe it would be best to expose a 3rd argument, leaving first and second argument as it is:
- Initial state
- State creators / mutators
- Function to merge state, state mutators and `setState` to props

Thoughts?